### PR TITLE
[WiP] feat: add market price insights picker

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **Feature**

### Description
- Add market price insights picker

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [ ] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434

### CODE REVIEW

File: src/lib/Components/Artist/ArtistInsights/MarketStats.tsx [hunk 1]
```diff
@@ -147,8 +173,8 @@ export const MarketStatsQueryRenderer: React.FC<{
       variables={{ artistInternalID }}
       query={graphql`
         query MarketStatsQuery($artistInternalID: ID!) {
-          priceInsights(artistId: $artistInternalID) {
-            ...MarketStats_priceInsights
+          priceInsightsConnection: priceInsights(artistId: $artistInternalID, sort: DEMAND) {
```
Comment (pepopowitz) [inline src/lib/Components/Artist/ArtistInsights/MarketStats.tsx:176]:
```suggestion
          priceInsightsConnection: priceInsights(artistId: $artistInternalID, sort: DEMAND_RANK_DESC) {
```

I think this is the correct sort based on the schema.

**Review by pepopowitz — APPROVED** [review]
Looks great!

**Comment by pepopowitz** [comment]
I fixed up the type error on the `sort` argument, but I'm going to reassign to you -- with WIP in the title I wasn't sure what you wanted to do yet before merging.

File: src/lib/Components/Artist/ArtistInsights/MarketStats.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/lib/Components/Artist/ArtistInsights/MarketStats.tsx]: thanks for fixing this

File: src/lib/Components/Artist/ArtistInsights/MarketStats.tsx [hunk 2]
```diff
@@ -1,24 +1,43 @@
...
 
+  const priceInsights = extractNodes(priceInsightsConnection)
+
+  if (priceInsights.length === 0) {
+    return null
+  }
+
+  const [selectedPriceInsight, setSelectedPriceInsight] = useState(priceInsights[0])
+
+  const mediumOptions = useRef<Array<{ value: string; label: string }>>(
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/lib/Components/Artist/ArtistInsights/MarketStats.tsx:30]: I just wanted a list of the medium options that won't be computed again if the component rerendered. I could have also used `useCallback` or `useMemo`

[the remark this answers is not in the document]
Comment (pepopowitz) [reply src/lib/Components/Artist/ArtistInsights/MarketStats.tsx:30]: ah, I see. I think `useMemo` would have been more clear --- to me at least, `useRef` implies there's some interaction with the DOM outside of React.

**Pull request merged**